contract_manager/src/contracts/near.ts

```diff
@@ -0,0 +1,264 @@
+import { DataSource } from "@pythnetwork/xc-admin-common";
+import {
+  KeyValueConfig,
+  PriceFeed,
+  PriceFeedContract,
+  PrivateKey,
+  TxResult,
+} from "../base";
+import { Chain, NearChain } from "../chains";
+import * as nearAPI from "near-api-js";
+import { BN } from "fuels";
+import { WormholeContract } from "./wormhole";
+
+export class NearWormholeContract extends WormholeContract {
+  static type = "NearWormholeContract";
+
+  constructor(public chain: NearChain, public address: string) {
+    super();
+  }
+
+  getId(): string {
+    return `${this.chain.getId()}__${this.address.replace(/-|\./g, "_")}`;
+  }
+
+  getChain(): NearChain {
+    return this.chain;
+  }
+
+  getType(): string {
+    return NearWormholeContract.type;
+  }
+
+  static fromJson(
+    chain: Chain,
+    parsed: { type: string; address: string }
+  ): NearWormholeContract {
+    if (parsed.type !== NearWormholeContract.type)
+      throw new Error("Invalid type");
+    if (!(chain instanceof NearChain))
+      throw new Error(`Wrong chain type ${chain}`);
+    return new NearWormholeContract(chain, parsed.address);
+  }
+
+  toJson(): KeyValueConfig {
+    return {
+      chain: this.chain.getId(),
+      address: this.address,
+      type: NearWormholeContract.type,
+    };
+  }
+
+  async upgradeGuardianSets(
+    senderPrivateKey: PrivateKey,
+    vaa: Buffer
+  ): Promise<TxResult> {
+    const senderAddress = await this.chain.getAccountAddress(senderPrivateKey);
+    const account = await this.chain.getNearAccount(
+      senderAddress,
+      senderPrivateKey
+    );
+    const outcome = await account.functionCall({
+      contractId: this.address,
+      methodName: "submit_vaa",
+      args: { vaa: vaa.toString("hex") },
+      gas: new BN(300e12),
+      attachedDeposit: new BN(1e12),
+    });
+    return { id: outcome.transaction.hash, info: outcome };
+  }
+
+  getCurrentGuardianSetIndex(): Promise<number> {
```

Comment:
so there is no way to check this out? strange.

Reply:
Yes, there is no method for this.